plugins/statistics.koplugin/main.lua

```diff
@@ -1,32 +1,41 @@
-local Widget = require("ui/widget/widget")
-local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local DocSettings = require("docsettings")
+local Event = require("ui/event")
+local InfoMessage = require("ui/widget/infomessage")
 local KeyValuePage = require("ui/widget/keyvaluepage")
-local UIManager = require("ui/uimanager")
-local Screen = require("device").screen
+local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ReaderProgress = require("readerprogress")
+local ReadHistory = require("readhistory")
+local SQ3 = require("lua-ljsqlite3/init")
 local TimeVal = require("ui/timeval")
-local DataStorage = require("datastorage")
+local UIManager = require("ui/uimanager")
+local Widget = require("ui/widget/widget")
 local lfs = require("libs/libkoreader-lfs")
-local DEBUG = require("dbg")
-local T = require("ffi/util").template
-local joinPath = require("ffi/util").joinPath
-local _ = require("gettext")
+local logger = require("logger")
 local util = require("util")
-local tableutil = require("tableutil")
-local ReadHistory = require("readhistory")
-local DocSettings = require("docsettings")
-local ReaderProgress = require("readerprogress")
-local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local _ = require("gettext")
+local joinPath = require("ffi/util").joinPath
+local Screen = require("device").screen
+local T = require("ffi/util").template
 
--- a copy of page_max_read_sec
-local page_max_time
+local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local db_location = DataStorage:getSettingsDir() .. "/book.sqlite3"
 
 local ReaderStatistics = Widget:extend{
-    last_time = nil,
-    page_min_read_sec = 5,
-    page_max_read_sec = 90,
-    current_period = 0,
-    pages_current_period = 0,
+    page_min_read_sec = 1,
+    page_max_read_sec = 180,
```

Comment:
I timed it. On this page with difficult French 60s is only enough to finish the top two paragraphs. Now I'm more interested in when there was reading activity than accurate reading time stats but clearly depending on what and how you're reading 3 minutes isn't some outrageous value.

![p_20170514_211611](https://cloud.githubusercontent.com/assets/202757/26037024/e05da9e8-38ea-11e7-83c7-b440f2eacc69.jpg)